models/user.ts

```diff
@@ -0,0 +1,17 @@
+import { model, Schema } from 'mongoose';
+import { IUser } from '../types/user';
+
+// A logged-in User to Husky Habits
+const userSchema: Schema = new Schema<IUser>(
+  {
+    email: { type: String, required: true },
+    first_name: String,
+    last_name: String,
+    accounts: [{ acc_type: String, uid: String }],
+    deleted: { type: Boolean, default: false },
```

Comment:
How would this be used? Say if we delete a user and want to keep storing their data?

Reply:
also - does the profile need a soft delete as well? maybe we could pull that out as a separate "BaseSchema" class which all of our schemas are derived from.

Reply:
I personally don't love having every possible schema be a soft delete; I think we should only introduce a soft delete when we explicitly have a "recover" option for the user because it's otherwise unnecessary

I definitely do agree with having a soft delete on the profiles (and I think that's originally what I wanted but mixed them up lol), but maybe a hard delete on a user? It seems like a lot to store a deleted user and their deleted profiles imo, as a user can just re-register with the same email without their profiles, but let me know if you disagree